### PR TITLE
TLDR-386 pdf auto reader bug

### PR DESCRIPTION
**Pull request by NastyBoget**
- refactor pdf_txtlayer_classifier, its training script and benchmarking;
- added script for training data generation;
- generate more training data;
- changed classifier from xgboost to catboost, added feature importances;
- uploaded data and the new classifier on cloud.

### CODE REVIEW

File: dedoc/scripts/train/train_txtlayer_classifier.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import zipfile
+from pathlib import Path
+from typing import List, Tuple
+
+import wget
+import xgbfir
+from sklearn.metrics import f1_score
+from xgboost import XGBClassifier
+
+from config import get_config
```
Comment (oksidgy) [inline dedoc/scripts/train/train_txtlayer_classifier.py:13]: from dedoc.config

File: dedoc/scripts/train/train_txtlayer_classifier.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    def __make_path(self, path: Path) -> List[str]:
+        if not path.is_dir():
+            return []
+
+        path_all = []
+        for subdir in path.iterdir():
+            if not subdir.is_dir():
+                continue
+            for file_path in subdir.iterdir():
+                if str(file_path).endswith("txt"):
```
Comment (oksidgy) [inline dedoc/scripts/train/train_txtlayer_classifier.py:35]: endswith(".txt")

File: dedoc/scripts/train/train_txtlayer_classifier.py (hunk 1 above)
Comment (NastyBoget) [reply dedoc/scripts/train/train_txtlayer_classifier.py]: done

File: dedoc/scripts/train/train_txtlayer_classifier.py (hunk 2 above)
Comment (NastyBoget) [reply dedoc/scripts/train/train_txtlayer_classifier.py]: done

**Pull request merged**